packages/core/src/behaviors.ts

```diff
@@ -0,0 +1,123 @@
+import { ActorContext, ActorRef, Behavior, EventObject, Observer } from '.';
+import { doneInvoke, error } from './actions';
+import { toActorRef } from './Actor';
+import { toObserver } from './utils';
+
+/**
+ * Returns an actor behavior from a reducer and its initial state.
+ *
+ * @param transition The pure reducer that returns the next state given the current state and event.
+ * @param initialState The initial state of the reducer.
+ * @returns An actor behavior
+ */
+export function fromReducer<TState, TEvent extends EventObject>(
+  transition: (
+    state: TState,
+    event: TEvent,
+    actorContext: ActorContext<TEvent, TState>
+  ) => TState,
+  initialState: TState
+): Behavior<TEvent, TState> {
+  return {
+    transition,
+    initialState
+  };
+}
+
+type PromiseEvents<T> =
+  | { type: 'fulfill'; data: T }
+  | { type: 'reject'; error: unknown };
+
+export function fromPromise<T>(
+  promiseFn: () => Promise<T>
+): Behavior<PromiseEvents<T>, T | undefined> {
+  return {
+    transition: (state, event, { parent, id, observers }) => {
+      switch (event.type) {
+        case 'fulfill':
+          parent?.send(doneInvoke(id, event.data));
+          return event.data;
+        case 'reject':
+          parent?.send(error(id, event.error));
+          observers.forEach((observer) => {
+            observer.error(event.error);
+          });
```

Comment:
it seems a little bit quirky that observers are called here in the error case while they are called from within `flush` for the next item case

I'm not sure how to change this atm though